docs/DeveloperGuide.md

```diff
@@ -262,43 +320,52 @@ complexity.
 
 ## Non-Functional Requirements
 
-* Technical Requirements: Any mainstream OS, i.e. Windows, macOS or Linux, with Java 11 installed. Instructions for downloading Java 11 can be found [here](https://www.oracle.com/sg/java/technologies/javase/jdk11-archive-downloads.html).
-* Project Scope Constraints: The application should only be used for tracking. It is not meant to be involved in any form of monetary transaction.
+* Technical Requirements: Any *mainstream OS* with Java 17 or above installed. 
+  Instructions for downloading Java 17 can be found 
+  [here](https://www.oracle.com/sg/java/technologies/javase/jdk17-archive-downloads.html).
+* Project Scope Constraints: The application should only be used for tracking. It is not meant to be involved in any
+  form of monetary transaction.
 * Project Scope Constraints: Data storage is only to be performed locally.
-* Quality Requirements: The application should be able to be used effectively by a novice with little experience with CLIs.
+* Quality Requirements: The application should be able to be used effectively by a novice with little experience with
+  CLIs.
 
 ## Glossary
 
-[ADD GLOSSARY HERE]
+**Mainstream OS**: Windows, Linux, Unix, MacOS
```

Comment:
Perhaps this section has other terms you can consider adding in.

![image](https://github.com/user-attachments/assets/7c5deb81-ea85-41e9-ab66-dade189b9148)